virtualmachine/vsphere/util.go

```diff
@@ -1200,15 +1200,21 @@ var reconfigureVM = func(vm *VM, vmMo *mo.VirtualMachine) error {
 
 var waitForIP = func(vm *VM, vmMo *mo.VirtualMachine) error {
 	vmObj := object.NewVirtualMachine(vm.client.Client, vmMo.Reference())
-	ipString, err := vmObj.WaitForIP(vm.ctx)
+	// second parameter is to list v4 ips only and ignore v6 ips
+	ipMap, err := vmObj.WaitForNetIP(vm.ctx, true)
 	if err != nil {
 		return fmt.Errorf("failed to wait for VM to boot up: %v", err)
```

Comment:
Should this error be "failed to wait for VM to get ips...".

Reply:
this is not required now. Changed the code a bit.

Reply:
We are still returning err at line no 1206. And we are returning error "failed to wait for VM to boot up: %v" though we get this error while waiting for ip.

Reply:
oh okay, missed it.